src/systems/menu_impl.cpp

```diff
@@ -66,68 +70,55 @@ static bool is_widget_selectable(WidgetType type) {
            type == WidgetType::InputSelect;
 }
 
-static Widget* get_sub_selected_widget(HideableGroupWidget* widget, s32* selectable, s32* sel) {
-    for (u32 i = 0; i < widget->num_widgets; i++) {
-        Widget* child = &widget->widgets[i];
-        if (is_widget_selectable(child->type)) {
-            (*selectable)++;
-            if (*selectable == *sel) return child;
-        } else if (child->type == WidgetType::HideableGroupWidget && show_hideable_widget(child)) {
-            Widget* possible_selection =
-                get_sub_selected_widget(&child->hideable_group, selectable, sel);
-            if (possible_selection != nullptr) {
-                return possible_selection;
-            }
-        }
+// declare here so it can be used in get_selected_widget functions
+static Widget* find_widget(Widget* widgets, s32& selectable, s32 sel);
+
+static Widget* get_sub_selected_widget(HideableGroupWidget* hideable, s32& curr_sel_idx,
+                                       s32 target_sel_idx) {
+    for (u32 i = 0; i < hideable->num_widgets; i++) {
+        Widget* result = find_widget(&hideable->widgets[i], curr_sel_idx, target_sel_idx);
+        if (result != nullptr) return result;
     }
 
     return nullptr;
 }
```

Comment:
Ah, interesting ping-pong recursion approach! I was originally thinking of a single function taking a widget list could directly recurse on either a `MenuWidget`'s or `HideableGroupWidget`'s sub-widgets but this works too.